ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -14033,12 +14033,26 @@ ABoundActivity * HqlCppTranslator::doBuildActivityDedup(BuildCtx & ctx, IHqlExpr
 
     if (!useHash)
     {
-        if (info.compareAllRows)
-            doBuildBoolFunction(instance->classctx, "compareAll", info.compareAllRows);
-        if (!info.keepLeft)
-            doBuildBoolFunction(instance->classctx, "keepLeft", info.keepLeft);
+        // Only necessary to create getFlags() if different from base class getFlags() - i.e. different from default
+        if ( !(!info.compareAllRows && !info.keepBest && info.keepLeft) )
+        {
+            StringBuffer flags;
+            if (info.compareAllRows)
+                flags.append("|DFcompareall");
+            if (info.keepLeft)
+                flags.append("|DFkeepleft");
+            if (info.keepBest)
+                flags.append("|DFkeepbest");
+            if (flags.length()==0)
+                flags.append("|0");
+            doBuildUnsignedFunction(instance->startctx, "getFlags", flags.str()+1);
+        }
         if (!matchesConstantValue(info.numToKeep, 1))
+        {
             doBuildUnsignedFunction(instance->startctx, "numToKeep", info.numToKeep);
+            if (info.keepBest)
+                throwError(HQLERR_DedupBestWithKeepn);
```

Comment:
>I noticed that there is no current support in the case of keepLeft=False and numToKeep>1.  I assumed this is because the performance would be adversely affected because 'numToKeep' rows will need to be cached in memory.

That's true, but we do similar elsewhere (for keep in join I think). As long as N is reasonable (we could put a cap on it, e.g 10k (I think we do that elsewhere too), then it's not prohibitively expensive.

The real reason it's not supported is probably lack of demand, and/or other priorities.

I think BEST, KEEP(n) combination would be more useful though..